tensorflow_quantum/core/qsim/util.cc

```diff
@@ -33,5 +35,21 @@ void* _aligned_malloc(size_t size) {
 
 void _aligned_free(void* ptr) { free(*(reinterpret_cast<void**>(ptr) - 1)); }
 
+int ComputeParity(const absl::flat_hash_set<unsigned int>& parity_set, const uint64_t sample) {
+  uint64_t sample_copy(sample);
+  int parity(1);
+  unsigned int location = 0;
+  while (sample_copy > 0) {
+    if (sample_copy % 2 == 1) {
+      if (parity_set.find(location) != parity_set.end()) {
+        parity *= -1;
+      }
+    }
+    location++;
+    sample_copy /= 2;
+  }
+  return parity;
+}
```

Comment:
Can this be simplified to something like this ?
```C++
inline int parity(const std::vector<int>& measured_bits, const uint64_t bitstring) {
	uint64_t mask = 0;
	for(int i =0;i<measured_bits.size();i++){
		mask |= uint64_t(1) << uint64_t(measured_bits[i]);
	}
	int count = std::bitset<64>(bitstring & mask).count() & 1;
	return count ? -1 : 1;
}
```
since the ith bit in `bitstring` will be set if qubit i measured 1 ?

Reply:
Good point, this is a perfect candidate for bit operations, I've now put this code into the header